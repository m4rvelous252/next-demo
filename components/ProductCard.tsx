import { Product } from "@/app/[locale]/product/ApiProduct";
import Image from "next/image";

interface ProductCardProps {
	product: Product;
}

const ProductCard = ({ product }: ProductCardProps) => {
	return (
		<div
			data-testid='product-card'
			className='border p-6 rounded shadow flex flex-col'>
			<h3 className='text-xl font-semibold'>{product.title}</h3>
			<h6 className='mb-4'>{product.brand}</h6>
			{product.thumbnail ? (
				<div className='relative w-full h-48 rounded-sm overflow-hidden'>
					<Image
						alt={product.title}
						src={product.thumbnail}
						fill
						className='object-cover'
					/>
				</div>
			) : null}
			<p className='mt-4'>
				{product.price.toLocaleString("en", {
					style: "currency",
					currency: "USD",
				})}
			</p>
			<p className='mt-2'>{product.description}</p>
		</div>
	);
};

export default ProductCard;
