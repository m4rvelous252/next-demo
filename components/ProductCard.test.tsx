import { expect } from "@jest/globals";
import { render, screen } from "@testing-library/react";
import ProductCard from "./ProductCard";

const mockProduct = {
	id: 1,
	title: "Test Product",
	brand: "Test Brand",
	thumbnail:
		"https://plus.unsplash.com/premium_photo-1679470210717-97fc30968fdf?q=80&w=1974&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
	price: 9.99,
	description: "Test description",
};
test("matches snapshot", () => {
	const { asFragment } = render(<ProductCard product={mockProduct} />);
	expect(asFragment()).toMatchSnapshot();
});

test("renders product card with correct data", () => {
	render(<ProductCard product={mockProduct} />);

	// Assert that the product title is rendered
	expect(screen.getByText("Test Product")).toBeInTheDocument();

	// Assert that the product brand is rendered
	expect(screen.getByText("Test Brand")).toBeInTheDocument();

	// Assert that the product image is rendered
	expect(screen.getByAltText("Test Product")).toBeInTheDocument();

	// Assert that the product price is rendered
	expect(screen.getByText("$9.99")).toBeInTheDocument();

	// Assert that the product description is rendered
	expect(screen.getByText("Test description")).toBeInTheDocument();
});

test("renders product without thumbnail", () => {
	render(<ProductCard product={{ ...mockProduct, thumbnail: undefined }} />);

	// Assert that the product image is not rendered
	expect(screen.queryByAltText("Test Product")).not.toBeInTheDocument();
});
