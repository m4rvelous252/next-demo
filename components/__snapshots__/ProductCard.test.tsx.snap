// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="border p-6 rounded shadow flex flex-col"
    data-testid="product-card"
  >
    <h3
      class="text-xl font-semibold"
    >
      Test Product
    </h3>
    <h6
      class="mb-4"
    >
      Test Brand
    </h6>
    <div
      class="relative w-full h-48 rounded-sm overflow-hidden"
    >
      <img
        alt="Test Product"
        class="object-cover"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=https%3A%2F%2Fplus.unsplash.com%2Fpremium_photo-1679470210717-97fc30968fdf%3Fq%3D80%26w%3D1974%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fplus.unsplash.com%2Fpremium_photo-1679470210717-97fc30968fdf%3Fq%3D80%26w%3D1974%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fplus.unsplash.com%2Fpremium_photo-1679470210717-97fc30968fdf%3Fq%3D80%26w%3D1974%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fplus.unsplash.com%2Fpremium_photo-1679470210717-97fc30968fdf%3Fq%3D80%26w%3D1974%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fplus.unsplash.com%2Fpremium_photo-1679470210717-97fc30968fdf%3Fq%3D80%26w%3D1974%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fplus.unsplash.com%2Fpremium_photo-1679470210717-97fc30968fdf%3Fq%3D80%26w%3D1974%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fplus.unsplash.com%2Fpremium_photo-1679470210717-97fc30968fdf%3Fq%3D80%26w%3D1974%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fplus.unsplash.com%2Fpremium_photo-1679470210717-97fc30968fdf%3Fq%3D80%26w%3D1974%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fplus.unsplash.com%2Fpremium_photo-1679470210717-97fc30968fdf%3Fq%3D80%26w%3D1974%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </div>
    <p
      class="mt-4"
    >
      $9.99
    </p>
    <p
      class="mt-2"
    >
      Test description
    </p>
  </div>
</DocumentFragment>
`;
